src/main.cpp

```diff
@@ -310,11 +322,26 @@ int main(int argc, char * argv[])
     if (!(ErrorDialogHandler::instance()->checkError())) {
         qDebug() << "Displaying mixxx";
         mixxx->show();
+        splash.finish(mixxx);
 
         qDebug() << "Running Mixxx";
         result = a.exec();
     }
 
+    QMessageBox cleaningUpBox;
+    cleaningUpBox.setStandardButtons(0);
+    cleaningUpBox.setWindowTitle(QObject::tr("Exiting Mixxx"));
+    cleaningUpBox.setText(QObject::tr(
+            "Cleaning up the library. This may take some time.\n"
+            "Do not restart Mixxx or shutdown the device until\n "
+            "this box has disappeared."));
```

Comment:
In general, I feel that splash screens and shutdown screens are symptoms of problems and are not solutions. It's too easy for users to ignore the text and still trigger the bugs.  There are some real bugs you have identified, and I think we should fix those bugs rather than hope users read the text.

It sounds like we need to implement better DB locking so that two instances of mixxx cannot run at the same time.  Then we don't need any text at all -- if the user tries to start a second mixxx while the first is shutting down, the second mixxx will just block until the first one exits.  (After a timeout, we could display text: "Unable to start because it appears that Mixxx is already running. [link to wiki explanation?]")  

As for forcibly turning off the PC power while mixxx is exiting or killing the process manually, I don't think text will help there -- it's not mixxx's job to educate users on basic computer operation.  Users should know to use the menu or power button to properly shut off their machine.  A better way to protect against this would be to make a backup copy of the DB on startup if the DB loads correctly.  That way if anything goes wrong there will be a last-known-good version of the DB that the user can manually restore from.